lib/variable/reduction.cpp

```diff
@@ -206,6 +203,76 @@ Variable any(const Variable &var) {
   return reduce_all_dims(var, [](auto &&... _) { return any(_...); });
 }
 
+/// Return the mean along all dimensions.
+Variable mean(const Variable &var) {
+  return normalize_impl(sum(var), count(var));
+}
+
+/// Return the mean along all dimensions. Ignoring NaN values.
+Variable nanmean(const Variable &var) {
+  return normalize_impl(nansum(var), sum(isfinite(var)));
+}
+
+/// Return the sum of all events per bin.
+Variable bins_sum(const Variable &data) {
+  return reduce_bins(data, variable::sum_into, FillValue::ZeroNotBool);
+}
+
+/// Return the sum of all events per bin. Ignoring NaN values.
+Variable bins_nansum(const Variable &data) {
+  return reduce_bins(data, variable::nansum_into, FillValue::ZeroNotBool);
+}
+
+/// Return the maximum of all events per bin.
+Variable bins_max(const Variable &data) {
+  return reduce_bins(data, variable::max_into, FillValue::Lowest);
+}
+
+/// Return the maximum of all events per bin. Ignoring NaN values.
+Variable bins_nanmax(const Variable &data) {
+  return reduce_bins(data, variable::nanmax_into, FillValue::Lowest);
+}
+
+/// Return the minimum of all events per bin.
+Variable bins_min(const Variable &data) {
+  return reduce_bins(data, variable::min_into, FillValue::Max);
+}
+
+/// Return the minimum of all events per bin. Ignoring NaN values.
+Variable bins_nanmin(const Variable &data) {
+  return reduce_bins(data, variable::nanmin_into, FillValue::Max);
+}
+
+/// Return the logical AND of all events per bin.
+Variable bins_all(const Variable &data) {
+  return reduce_bins(data, variable::all_into, FillValue::True);
+}
+
+/// Return the logical OR of all events per bin.
+Variable bins_any(const Variable &data) {
+  return reduce_bins(data, variable::any_into, FillValue::False);
+}
+
+namespace {
+Variable bin_sizes_without_mask(const Variable &data) {
+  if (const auto binned_mask = binned_irreducible_mask(data);
+      binned_mask.is_valid()) {
+    return bins_sum(binned_mask);
+  }
+  return bin_sizes(data);
+}
+} // namespace
```

Comment:
For consistency with above, this should be named `bins_count`? May also be worth placing it side by side for readability.